src/lava/proc/rf/models.py

```diff
@@ -0,0 +1,121 @@
+
+        old_real = deepcopy(self.real)
+        old_imag = deepcopy(self.imag)
+        self.real = self.cos_decay * old_real - self.sin_decay * old_imag + a_real_in_data  # I don't think it needs a bias for now
+        self.imag = self.sin_decay * old_real + self.cos_decay * old_imag + a_imag_in_data
+
+    def run_spk(self):
+        old_imag = deepcopy(self.imag)
+        self.sub_thresh_dynamics()
+        s_out = (self.real >= self.vth)  * (self.imag >= 0) * (old_imag < 0)
+        self.s_out.send(s_out)
+
+
+
+
+@implements(proc=RF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyRFModelFloat(AbstractPyRFModelFloat):
+    pass
+
+class AbstractPyRFModelFixed(PyLoihiProcessModel):
+    a_real_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, np.int16, precision=16)
+    a_imag_in: PyInPort = LavaPyType(PyInPort.VEC_DENSE, np.int16, precision=16)
+    s_out: None
+    real: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    imag: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+    sin_decay: int = LavaPyType(int, np.uint16, precision=12)   # 0 4095
+    cos_decay:int = LavaPyType(int, np.uint16, precision=12)   # 0 to 4095
+    vth: np.ndarray = LavaPyType(np.ndarray, np.int32, precision=24)
+
+    def __init__(self, proc_params):
+        super(AbstractPyRFModelFixed, self).__init__(proc_params)
+        
+        self.decay_shift = 12
+        self.ri_bitwidth = 24 # real, imaginary bitwidth
+        self.max_ri_val = 2 ** (self.ri_bitwidth - 1)
+        self.act_shift = 6
+        self.vth_shift = 6
+
+        self.c_offset = 1
+        self.s_offset = 1
+
+    def sub_thresh_dynamics(self):
+
+        decay_const_cos = self.cos_decay + self.c_offset
+        decay_const_sin = self.sin_decay + self.s_offset
+
+        a_real_in_data = self.a_real_in.recv()
+        a_imag_in_data = self.a_imag_in.recv()
+
+
+        decayed_real = np.int64(self.real) * (decay_const_cos) - np.int64(self.imag) * (decay_const_sin)
+        decayed_real = np.sign(decayed_real) * np.right_shift(np.abs(decayed_real), self.decay_shift)
+        decayed_real = np.int32(decayed_real)
+
+        a_real_in_data = np.left_shift(a_real_in_data, self.act_shift)
+        decayed_real += a_real_in_data
+
+        decayed_imag = np.int64(self.real) * ( decay_const_sin) + np.int64(self.imag) * (decay_const_cos)
+        decayed_imag = np.sign(decayed_imag) * np.right_shift(np.abs(decayed_imag), self.decay_shift)
+        decayed_imag = np.int32(decayed_imag)
+        
+        a_imag_in_data = np.left_shift(a_imag_in_data, self.act_shift)
+        decayed_imag += a_imag_in_data
+
+        neg_voltage_limit = -np.int32(self.max_ri_val) + 1
+        pos_voltage_limit = np.int32(self.max_ri_val) - 1
+
+        self.real[:] = np.clip(decayed_real, neg_voltage_limit, pos_voltage_limit)
+        self.imag[:] = np.clip(decayed_imag, neg_voltage_limit, pos_voltage_limit)
+        
+    def scale_threshold(self):
```

Comment:
Let's remove this scaling option. Instead, the value in the process should be scaled directly.